### DISCUSSION TITLE
Build and Start Workflow from JSON Definition

### DISCUSSION THREAD

**Discussion by jamshally** [elsa-workflows/elsa-core#2750] [Q&A]
It it possible to 'Build and Start' a Workflow from a JSON definition file?  

Apologies if this is already answered in the documentation, but I couldn't see it.

**Comment by sfmskywalker** [comment]
Most certainly. The recommended way is to register the `BlobStorageWorkflowProvider`  like this:

```csharp
// Configure blob storage for blob storage workflow storage provider. 
services.Configure<BlobStorageWorkflowProviderOptions>(options => options.BlobStorageFactory = () => StorageFactory.Blobs.DirectoryFiles(Path.Combine(_environment.ContentRootPath, "Workflows")));
```

Where "Workflows" would be a directory you create in the root of your project and store workflow JSON files.
These files then automatically become available to the system, which means you can use the workflow registry to find & run workflows for example.

A more "immediate" mode is possible too by doing these steps:

1. Load & parse a JSON file into a workflow definition.
2. Create a workflow blueprint from the workflow definition.
3. Run the workflow.

Example:

```csharp

// Read JSON from some file.
var json = File.ReadAllTextAsync("some-workflow.json");

// Parse JSON into a WorkflowDefinition.
var model = _contentSerializer.Deserialize<WorkflowDefinition>(json); // IContentSerializer

// Materialize a workflow blueprint. This is what we can execute.
var blueprint = await _workflowBlueprintMaterializer.CreateWorkflowBlueprintAsync(workflowDefinition); // IWorkflowBlueprintMaterializer

// Run the workflow.
var result = await _workflowStarter.StartWorkflowAsync(blueprint); // IStartsWorkflow
```
